### ISSUE TITLE
Question around best practice to not need to drag and drop every time

### ISSUE THREAD

**Issue by j842** [p-jackson/grizzly-pass#3]
To avoid dragging and dropping every time you bring up your dashboard, is the intent that you include your own JSON together with the source code and host yourself?

I was wondering about being able to give people a URL that 'just works'. Two example solutions might be:

1) Specifying a URL for the JSON file in the query string.

2) New UI to enter a URL for the JSON file, and a feature to generate a permalink that includes that URL somehow.

It'd also mean you could just refresh the page to see changes to the JSON file. I'm not sure how this would work with access control though, if grizzly is giving you a dashboard with sensitive info in it. So it may be easier to just host everything and replace the JSON in the checkout... :-)

**Comment by p-jackson** [comment]
To fix the problem of having to re-drop the file every time you want to see your dashboard, my current idea is to have it save dashboards locally, so you'd just go to grizzly again and your dashboards would just be there. Shoebox app like. I think that works nicely if you just think of grizzly as a desktop app.

Sharing your dashboards with others is something I do want to support though. But because it involved deciding how sensitive the data in dashboards was, I was putting it in the later basket. For the public data case I'd like grizzly to be able to support publishing.

I really don't want to get into storing sensitive data though, so pointing grizzly at a url could get around that. The person hosting the data would have to do work to enable cross-origin requests. And there'd also be security considerations on grizzly's end.

**Comment by p-jackson** [comment]
What if there was a **save as static html** button? Then you can host that wherever you want and it'd be your problem to prevent your users getting hacked?

**Comment by j842** [comment]
Yes - I like the static html idea a lot! Could you programmatically make that happen?
(e.g. a travis task that runs when the JSON gets updated, which dumps out the static HTML and publishes it?)

**Comment by p-jackson** [comment]
Yes, but I think that'd be kind like `grizzly-pass-cli`

It could use the same code as the web project, but would be a node command-line tool.

________________________________

**Comment by p-jackson** [comment]
Yes, but I think that'd be kind like `grizzly-pass-cli`
It could use the same code as the web project, but would be a node command-line tool.

**Comment by j842** [comment]
Yeah - in that workflow the web project is great when you're writing the JSON, as it's an easier way to try it out then messing with a command line tool. :-)

A grizzly-pass-cli would be awesome.

**Comment by j842** [comment]
We worked around this with a custom fork that pulls JSON files from a hard-coded location (S3 bucket), choosing the JSON file based on info encoded in the URL. Ugly but it works :-)
